### PR TITLE
#7172

### PR DESCRIPTION
**Pull request by nbogol**
- fix EmailValidator
- added unit tests

### CODE REVIEW

**Comment by nbogol** [comment]
Not sure where all these errors are coming from. Most likely the test environment is not in a good shape.

**Comment by samdark** [comment]
#7172

File: tests/unit/framework/validators/EmailValidatorTest.php [hunk 1]
```diff
@@ -95,6 +95,16 @@ public function testValidateValueMx()
         $this->assertTrue($validator->validate('test@nonexistingsubdomain.example.com'));
         $validator->checkDNS = true;
         $this->assertFalse($validator->validate('test@nonexistingsubdomain.example.com'));
+
+        $validator->checkDNS = true;
```
Comment (samdark) [inline tests/unit/framework/validators/EmailValidatorTest.php]: Do we need to enable `checkDNS` in order to test it?

**Comment by cebe** [comment]
Merged, thanks!

**Pull request closed without merging**